src/main/java/de/zalando/aruha/nakadi/controller/EventTypeController.java

```diff
@@ -0,0 +1,103 @@
+package de.zalando.aruha.nakadi.controller;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.module.SimpleModule;
+import com.fasterxml.jackson.databind.node.TreeTraversingParser;
+import com.fasterxml.jackson.datatype.jdk8.Jdk8Module;
+import com.fasterxml.jackson.datatype.jsonorg.JSONObjectDeserializer;
+import com.fasterxml.jackson.datatype.jsonorg.JSONObjectSerializer;
+import de.zalando.aruha.nakadi.domain.EventType;
+import de.zalando.aruha.nakadi.domain.Problem;
+import de.zalando.aruha.nakadi.problem.InvalidJsonProblem;
+import de.zalando.aruha.nakadi.repository.EventTypeRepository;
+import org.everit.json.schema.Schema;
+import org.everit.json.schema.ValidationException;
+import org.everit.json.schema.loader.SchemaLoader;
+import org.json.JSONObject;
+import org.json.JSONTokener;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RestController;
+import org.zalando.problem.ProblemModule;
+
+import javax.ws.rs.core.Response;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+import java.util.Optional;
+
+import static java.util.Optional.empty;
+import static org.springframework.http.ResponseEntity.status;
+
+@RestController
+@RequestMapping(value = "/event_types")
+public class EventTypeController {
+
+    private ObjectMapper jsonMapper;
+
+    private EventTypeRepository repository;
+
+    private Schema schema;
+
+    @Autowired
+    public EventTypeController(EventTypeRepository repository, ObjectMapper jsonMapper) {
+        this.repository = repository;
+        this.jsonMapper = jsonMapper;
+
+        setupJsonMapper();
+        loadEventTypeJsonSchema();
+    }
+
+    @RequestMapping(method = RequestMethod.POST)
+    public ResponseEntity<?> createEventType(@RequestBody final String requestBodyString) throws Exception {
+        try {
+            schema.validate(new JSONObject(requestBodyString));
```

Comment:
@v-stepanov yes, that's true. Can you imagine a scenario where that would be a problem?